### PR TITLE
UniverSC aligner

### PR DESCRIPTION
**Pull request by TomKellyGenetics**
# adds UniverSC aligner

Implements similar functionality to Cell Ranger v7.0.0 with the following exceptions:

- runs an open-source implementation for Cell Ranger v3.0.2
- is not subject to the 10X Genomics EULA and can be run on any technology on open-source Licenses (GPL3 and MIT)
- supports multiple single-cell platforms other than 10X Chromium (currently 40 technologies are supported)

Uses the module developed here:
https://github.com/nf-core/modules/pull/1706

With this container in Docker or Singularity:
https://hub.docker.com/r/nfcore/universc

Addresses issue #170 

Method discussed here in more detail: https://github.com/nf-core/scrnaseq/issues/31#issuecomment-772240225

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/scrnaseq/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/scrnaseq _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [x] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [x] `README.md` is updated (including new tool citations and authors/contributors).

## Unit tests

Testing incomplete (pending merge of UniverSC module) to nf-core/modules.

### CODE REVIEW

**Comment by apeltzer** [comment]
HI @TomKellyGenetics - this needs to be updated with your module from the central modules repository now :)

**Comment by TomKellyGenetics** [comment]
Thanks for your support to set up the docker image and module! @apeltzer I really appreciate your help to build this.

Thank you for the reminder as well. This is indeed the next thing on my list for this project. Honestly I’ve been busy with childcare commitments and am exhausted this week so it will need more time. I’ll update this thread when I’ve had time to give the module a test run in this subworkflow.

**Comment by TomKellyGenetics** [comment]
@apeltzer I've migrated changes to the UniverSC module to call CellRanger "mkref" from the cellranger module as discussed in the nf-core/modules PR. I'll check the automated CI results later in case further debugging or documentation is necessary. This subworkflow should run very similarly to the "cellranger" settings with the additional 'technology' parameter which is documented here: https://github.com/minoda-lab/universc#supported-technologies

In other words, I think this PR is ready for review. Thank you for your support.

**Review by apeltzer — APPROVED** [review]

**Comment by apeltzer** [comment]
Guess this needs some lint fixes

**Comment by TomKellyGenetics** [comment]
Thanks @apeltzer I was surprised that the linting problems were fairly minor. The automated CI was helpful for debugging, I've added a local copy of the module and set up `ch_universc_technology` inputs to ensure automated checks pass with 10x test settings.

Let me know if any other changes are needed (such as consistent style). It appears to be ready to merge. :-)

**Comment by TomKellyGenetics** [comment]
Closes #170

**Comment by klkeys** [comment]
nudging @apeltzer : the linter is pleased with these efforts, ok to merge?

**Comment by TomKellyGenetics** [comment]
Thanks, as it is now reviewed and I am a member of the organisation I could merge it myself but I’ll hold off as someone more familiar with this repository should make sure I’ve not missed anything. Based on our past discussions for the modules repository, waiting a couple of days is typical for our busy schedules. I’ll post a reminder on this thread if I don’t hear back in a while.

**Comment by apeltzer** [comment]
Yeah, we initially planned to have more maintenance releases between adding new features as the above but as linting works and tests are running I'm also not against it 👍🏻 Will merge now

**Comment by apeltzer** [comment]
Please report if there are things not working !

**Comment by TomKellyGenetics** [comment]
Thank you, this was reasonably straightforward compared to setting up the module. I think it is stable for now but I'll open a new PR if any further changes need to be considered. As discussed with @Emiller88 elsewhere I am willing to support maintaining this feature so please notify me if any relevant issues are opened by other users.
